pagopa/io-app
# chore: [IOPID-740] Zendesk token refresh handling

## Short description
This PR handles the Zendesk token refreshing during a FL session expiration.

## List of changes proposed in this pull request
* [ts/features/fastLogin/saga/tokenRefreshSaga.ts](https://github.com/pagopa/io-app/blob/f16110a4b83d6656eed8e8d2857fb9e09bb49b02/ts%2Ffeatures%2FfastLogin%2Fsaga%2FtokenRefreshSaga.ts)
* [ts/features/zendesk/saga/index.ts](https://github.com/pagopa/io-app/blob/f16110a4b83d6656eed8e8d2857fb9e09bb49b02/ts%2Ffeatures%2Fzendesk%2Fsaga%2Findex.ts): dismiss the support UI if it's open.
* [ts/features/zendesk/screens/ZendeskAskPermissions.tsx](https://github.com/pagopa/io-app/blob/f16110a4b83d6656eed8e8d2857fb9e09bb49b02/ts%2Ffeatures%2Fzendesk%2Fscreens%2FZendeskAskPermissions.tsx): add `zendeskGetSessionPollingLoop` saga to handle an eventual token expiration.
* [ts/features/zendesk/screens/ZendeskSeeReportsRouters.tsx](https://github.com/pagopa/io-app/blob/f16110a4b83d6656eed8e8d2857fb9e09bb49b02/ts%2Ffeatures%2Fzendesk%2Fscreens%2FZendeskSeeReportsRouters.tsx)
* [ts/features/zendesk/store/actions/index.ts](https://github.com/pagopa/io-app/blob/f16110a4b83d6656eed8e8d2857fb9e09bb49b02/ts%2Ffeatures%2Fzendesk%2Fstore%2Factions%2Findex.ts)
* [ts/utils/supportAssistance.ts](https://github.com/pagopa/io-app/blob/f16110a4b83d6656eed8e8d2857fb9e09bb49b02/ts%2Futils%2FsupportAssistance.ts): centralize duplicate code (DRY).
* [ts/features/zendesk/store/reducers/__tests__/index.test.ts](https://github.com/pagopa/io-app/blob/f16110a4b83d6656eed8e8d2857fb9e09bb49b02/ts%2Ffeatures%2Fzendesk%2Fstore%2Freducers%2F__tests__%2Findex.test.ts): fix tests.
* [ts/features/zendesk/store/reducers/index.ts](https://github.com/pagopa/io-app/blob/f16110a4b83d6656eed8e8d2857fb9e09bb49b02/ts%2Ffeatures%2Fzendesk%2Fstore%2Freducers%2Findex.ts): handle the polling lifecycle in the store.
* [ts/sagas/startup.ts](https://github.com/pagopa/io-app/blob/f16110a4b83d6656eed8e8d2857fb9e09bb49b02/ts%2Fsagas%2Fstartup.ts): start the Zendesk polling handler if the user is authenticated.
* [package.json](https://github.com/pagopa/io-app/blob/149a551ddeced929e35fd41711de8f60ebbdfd56/package.json)
* [yarn.lock](https://github.com/pagopa/io-app/blob/149a551ddeced929e35fd41711de8f60ebbdfd56/yarn.lock): bump Zendesk SDK version.

## How to test
Run the app against the dev server enabling fast login and the opt-in (optional):

```diff
diff --git a/src/payloads/backend.ts b/src/payloads/backend.ts
index dfc25ed..d3b54da 100644
--- a/src/payloads/backend.ts
+++ b/src/payloads/backend.ts
@@ -99,12 +99,12 @@ export const backendStatus: BackendStatus = {
     },
     fastLogin: {
       min_app_version: {
-        ios: "0.0.0",
+        ios: "1.0.0",
         android: "0.0.0"
       },
       opt_in: {
         min_app_version: {
-          ios: "0.0.0",
+          ios: "1.0.0",
           android: "0.0.0"
         }
       }
```
Then open a new support request and wait until the FL token expires. You should see the support UI being dismissed and the complimentary session expired screen being shown.

<details><summary>Example</summary>
<p>


<video src="https://github.com/pagopa/io-app/assets/16268789/e458b92f-6271-47d9-8a29-e3534a92b59b" />


</p>
</details>
